From 8f002ae789be715380c9aeb6f5d99c87504750d0 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Fri, 14 Dec 2012 13:05:34 +0200
Subject: [PATCH 3/4] OF: Compile Device Tree sources with resolve option

If the resolve config option has been enabled compile
the device tree sources with the appropriate flag to generate
the required for resolution __symbols__ node.

Note that both arch/arm/Makefile & arch/arm/boot/Makefile need
the DTC_FLAGS definition, because when issuing make dtbs the
boot Makefile doesn't seem to pick it.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 arch/arm/Makefile      | 6 +++++-
 arch/arm/boot/Makefile | 4 ++++
 scripts/Makefile.lib
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index cd87948..d1fae1e 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -286,6 +286,10 @@ ifeq ($(CONFIG_USE_OF),y)
 KBUILD_DTBS := dtbs
 endif
 
+ifeq ($(CONFIG_OF_RESOLVE),y)
+DTC_FLAGS += -@
+endif
+
 all:	$(KBUILD_IMAGE) $(KBUILD_DTBS)
 
 boot := arch/arm/boot
@@ -315,7 +319,7 @@ uImage-dtb.%:
 
 PHONY += dtbs
 dtbs: scripts
-	$(Q)$(MAKE) $(build)=$(boot)/dts MACHINE=$(MACHINE) dtbs
+	$(Q)$(MAKE) $(build)=$(boot)/dts MACHINE=$(MACHINE) DTC_FLAGS=$(DTC_FLAGS) dtbs
 
 # We use MRPROPER_FILES and CLEAN_FILES now
 archclean:
diff --git a/arch/arm/boot/Makefile b/arch/arm/boot/Makefile
index 4359029..ee6d0b2 100644
--- a/arch/arm/boot/Makefile
+++ b/arch/arm/boot/Makefile
@@ -60,6 +60,10 @@ $(obj)/zImage-dtb.%:	$(obj)/dts/%.dtb $(obj)/zImage
 
 endif
 
+ifeq ($(CONFIG_OF_RESOLVE),y)
+DTC_FLAGS += -@
+endif
+
 ifneq ($(LOADADDR),)
   UIMAGE_LOADADDR=$(LOADADDR)
 else
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 79c059e..efa7371 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -244,6 +244,10 @@ cmd_gzip = (cat $(filter-out FORCE,$^) | gzip -n -f -9 > $@) || \
 # DTC
 # ---------------------------------------------------------------------------
 
+ifeq ($(CONFIG_OF_RESOLVE),y)
+DTC_FLAGS += -@
+endif
+
 # Generate an assembly file to wrap the output of the device tree compiler
 quiet_cmd_dt_S_dtb= DTB     $@
 cmd_dt_S_dtb=                              
-- 
1.9.1

